cinema/urls.py

```diff
@@ -1,10 +1,26 @@
-from django.urls import path
+from django.urls import path, include
+from rest_framework import routers
+from cinema.views import MovieViewSet, GenreList, GenreDetail, CinemaHallViewSet
 
-from cinema.views import movie_list, movie_detail
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet)
 
+cinema_hall_list = CinemaHallViewSet.as_view(actions={"get": "list", "post": "create"})
+
+cinema_hall_detail = CinemaHallViewSet.as_view(
+    actions={
+        "get": "retrieve",
+        "put": "update",
+        "patch": "partial_update",
+        "delete": "destroy",
+    }
+)
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("", include(router.urls)),
+    path("genres/", GenreList.as_view(), name="genre_list"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genre-detail"),
+    path("cinema-hall", cinema_hall_list, name="cinema_hall_list"),
```

Comment:
Consider adding a trailing slash to the 'cinema-hall' path to follow RESTful API conventions. Change `"cinema-hall"` to `"cinema-hall/"`.